src/environment/EnvironmentManager.py

```diff
@@ -85,8 +89,38 @@ def run(env_name: str, *args, **kwargs) -> int:
         print(f"======== {agent_file} ========")
         res = system(
             f"conda run -n {EnvironmentManager.CONDA_ENV_PREFIX}_{env_name} "
+            + "--no-capture-output "
             + f"python {agent_file} {args_str} {kwargs_str}"
         )
         print(f"========={'=' * len(str(agent_file))}=========")
 
         return res == 0
+
+    @staticmethod
+    def send_latent_codes(latent_codes: np.ndarray) -> None:
```

Comment:
I would move port to a parameter with default value, i.e., "Send latent codes through socket via localhost using the specified port." instead. The default value could also be defined at the top of the file as a constant `DEFAULT_LATENT_CODE_PORT` or similar.

I believe that it is also better to describe the function's action, as in the example above